core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -260,26 +262,44 @@ public QueryValidationException validate() {
     }
 
     @Override
-    protected Query doToQuery(QueryParseContext parseContext) throws IOException {
+    protected Query doToQuery(QueryShardContext context) throws IOException {
         // Use the default field (_all) if no fields specified
         if (fieldsAndWeights.isEmpty()) {
-            String field = parseContext.defaultField();
+            String field = context.defaultField();
             fieldsAndWeights.put(field, 1.0F);
         }
 
+        // field names in builder can have wildcards etc, need to resolve them here
+        Map<String, Float> resolvedFieldsAndWeights = new TreeMap<>();
+        for (String fField : fieldsAndWeights.keySet()) {
+            if (Regex.isSimpleMatchPattern(fField)) {
+                for (String fieldName : context.mapperService().simpleMatchToIndexNames(fField)) {
+                    resolvedFieldsAndWeights.put(fieldName, fieldsAndWeights.get(fField));
+                }
+            } else {
+                MappedFieldType fieldType = context.fieldMapper(fField);
+                if (fieldType != null) {
+                    resolvedFieldsAndWeights.put(fieldType.names().indexName(), fieldsAndWeights.get(fField));
+                } else {
+                    resolvedFieldsAndWeights.put(fField, fieldsAndWeights.get(fField));
+                }
+            }
+        }
```

Comment:
This block moved here from the parser, because we need access to mapperService()